controller/src/controller/subsystems/instrument_comm.py

```diff
@@ -584,6 +652,17 @@ async def _process_command_response(self, packet_type: int, response_data: bytes
                 if self._firmware_update_manager is None:
                     raise NotImplementedError("_firmware_update_manager should never be None here")
                 await self._firmware_update_manager.update(command, response_data)
+            case "check_connection_status":
+                prev_command_info["status"] = response_data[0]
+                self._system_in_offline_mode = response_data[0] == ConnectionStatuses.HEADLESS.value
```

Comment:
You might not need the `.value` here, can't remember if `IntEnum` works that way or not though

Reply:
Yeah I used it in the pulse3d rewrite, requires `.value` or `int(ConnectionStatuses.HEADLESS)`

Reply:
It seems like with IntEnum that you don't need it
```
>>> test = bytes(range(3))
>>> 
>>> test
b'\x00\x01\x02'
>>> 
>>> test[1] == InstrumentConnectionStatuses.CONNECTED
True
>>> test[1] == InstrumentConnectionStatuses.OFFLINE
False
>>> test[0] == InstrumentConnectionStatuses.OFFLINE
False
>>> test[0] == InstrumentConnectionStatuses.DISCONNECTED
True
```

Reply:
`IntEnum` subclass `int` so that it inherits all the `int` behavior https://docs.python.org/3/library/enum.html#enum.IntEnum